packages/web-extension/src/messaging/remoteApi.ts

```diff
@@ -136,7 +139,8 @@ export const consumeMessengerRemoteApi = <T extends object>(
             takeUntil(unsubscribe$),
             filter(isEmitMessage),
             map(({ emit }) => emit),
-            shareReplay(1)
+            shareReplay(1),
+            filter(apiIsEnabled) // Do not replay values from an api object that was disabled
```

Comment:
This reads slightly better to me, as we're filtering the messages here
```suggestion
            filter(isNotDisabledApiMsg) // Do not replay values from an api object that was disabled
```

Reply:
Agree. Updated here: [307145f](https://github.com/input-output-hk/cardano-js-sdk/pull/556/commits/307145fec385223f7d8ca9fc3d46c89ba921aa4c)